File: front/src/app/solicitudes/solicitud.ts
/** Sedes donde se puede retirar el prestamo (enum Sede del backend). */
export const SEDES = ['SAN_JOAQUIN', 'PLAZA_OESTE', 'PUENTE_ALTO', 'VINA_DEL_MAR', 'CONCEPCION'] as const;
export type Sede = (typeof SEDES)[number];

/** Lo que se envia en POST /api/solicitudes (NuevaSolicitud.java). */
export interface NuevaSolicitud {
  items: { libroId: string; cantidad: number }[];
  dias: number;
  motivo: string;
  sede: Sede;
}

/** Una linea ya resuelta por el backend, con el titulo copiado del catalogo. */
export interface LineaSolicitud {
  libroId: string;
  titulo: string;
  autor: string;
  cantidad: number;
}

/** Lo que devuelve la API (Solicitud.java). */
export interface Solicitud {
  id: string;
  /** Sale del JWT (claim sub o client_id), no del formulario. */
  solicitante: string;
  estado: 'RECIBIDA' | 'APROBADA' | 'RECHAZADA';
  lineas: LineaSolicitud[];
  dias: number;
  motivo: string;
  sede: Sede;
  creadaEn: string;
  fechaDevolucion: string;
}

/** Cuerpo de error que devuelve ManejadorDeErrores del backend. */
export interface ErrorApi {
  status: number;
  error: string;
  mensaje: string;
  detalles: Record<string, string>;
}

File: back/src/main/java/cl/duoc/biblioteca/solicitudes/Solicitud.java
package cl.duoc.biblioteca.solicitudes;

import java.time.Instant;
import java.time.LocalDate;
import java.util.List;

/**
 * Una solicitud de prestamo ya creada. Es lo que devuelve la API.
 *
 * @param id               correlativo generado por el servidor, formato "SOL-0001"
 * @param solicitante      sacado del JWT (claim "sub" o "client_id"), NO del formulario
 * @param estado           ver {@link EstadoSolicitud}
 * @param lineas           libros pedidos, con su titulo resuelto desde el catalogo
 * @param dias             dias de prestamo solicitados
 * @param motivo           texto libre del formulario
 * @param sede             donde se retira
 * @param creadaEn         instante de creacion, en UTC
 * @param fechaDevolucion  creadaEn + dias; se calcula aqui para que el frontend no repita la regla
 */
public record Solicitud(
        String id,
        String solicitante,
        EstadoSolicitud estado,
        List<LineaSolicitud> lineas,
        int dias,
        String motivo,
        Sede sede,
        Instant creadaEn,
        LocalDate fechaDevolucion) {
}
